File: server/src/main/java/com/ivantrykosh/app/budgettracker/server/dtos/UserDto.java
package com.ivantrykosh.app.budgettracker.server.dtos;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.util.Date;

/**
 * User dto
 */
@NoArgsConstructor
@Getter
@Setter
@ToString
public class UserDto {
    private Long userId; // User ID

    private String email; // User email

    private String passwordHash; // Password hash

    private Date registrationDate; // Date of registration

    private Boolean isVerified; // Is user's email verified
}

File: client/app/src/main/java/com/ivantrykosh/app/budgettracker/client/data/remote/dto/UserDto.kt
package com.ivantrykosh.app.budgettracker.client.data.remote.dto

import com.ivantrykosh.app.budgettracker.client.domain.model.User
import java.util.Date

/**
 * Data class for user
 */
data class UserDto(
    val userId: Long,
    val email: String,
    val registrationDate: Date,
    val isVerified: Boolean,
)

/**
 * Convert from UserDto to User
 */
fun UserDto.toUser(): User {
    return User(
        userId = userId,
        email = email,
        registrationDate = registrationDate,
    )
}